pkg/api/plugin.go

```diff
@@ -23,9 +23,20 @@ type DeployFn func(context.Context, map[string]interface{}) error
 type PluginConfig struct {
 	SyncImage       string
 	AcceptLanguages []string
+	TestConfig      TestConfig // this should be empty in production
+}
+
+// TestConfig is embedded into PluginConfig
+type TestConfig struct {
+	RunningUnderTest   bool
+	ImageResourceGroup string
+	ImageResourceName  string
+	ImageOffer         string
+	ImageVersion       string
+	DeployOS           string
+	ORegURL            string
 }
 
-// Plugin is the main interface to openshift-azure
```

Comment:
we want all exported identifiers to have a docstring.  Improve it, but don't remove it.